frontend/src/bundles/profile/components/profile-settings/profile-settings.tsx

```diff
@@ -0,0 +1,232 @@
+import {
+    Avatar,
+    Button,
+    ButtonVariant,
+    DatePicker,
+    Input,
+    Loader,
+    Radio,
+} from '~/bundles/common/components/components.js';
+import { IconColor } from '~/bundles/common/components/icon/enums/enums.js';
+import { ComponentSize, Gender } from '~/bundles/common/enums/enums.js';
+import {
+    configureDateString,
+    configureISOString,
+    getObjectKeys,
+} from '~/bundles/common/helpers/helpers.js';
+import {
+    useAppForm,
+    useAppSelector,
+    useCallback,
+    useEffect,
+    useState,
+} from '~/bundles/common/hooks/hooks.js';
+import {
+    type UserUpdateProfileRequestDto,
+    userUpdateProfileValidationSchema,
+} from '~/bundles/users/users.js';
+
+import { DEFAULT_UPDATE_PROFILE_PAYLOAD } from './constants/constants.js';
+
+type Properties = {
+    onSubmit: (payload: UserUpdateProfileRequestDto) => void;
+    isLoading: boolean;
+};
+
+const ProfileSettings: React.FC<Properties> = ({ onSubmit, isLoading }) => {
+    const [valuesDefault, setValuesDefault] = useState(false);
+    const { user } = useAppSelector(({ auth }) => ({
+        user: auth.user,
+    }));
+
+    const { control, errors, reset, setValue, handleSubmit } =
+        useAppForm<UserUpdateProfileRequestDto>({
+            defaultValues: DEFAULT_UPDATE_PROFILE_PAYLOAD,
+            validationSchema: userUpdateProfileValidationSchema,
+            mode: 'onTouched',
+            shouldUnregister: false,
+        });
+
+    useEffect(() => {
+        if (!valuesDefault && user) {
+            for (const key of getObjectKeys(DEFAULT_UPDATE_PROFILE_PAYLOAD)) {
+                if (key === 'dateOfBirth' && user.dateOfBirth) {
+                    setValue(key, configureDateString(user.dateOfBirth));
+                } else {
+                    if (user[key]) {
+                        setValue(key, user[key]);
+                    } else {
+                        setValue(key, DEFAULT_UPDATE_PROFILE_PAYLOAD[key]);
+                    }
+                }
+            }
+            setValuesDefault(true);
+        }
+    }, [user, setValue, valuesDefault]);
+
+    const handleFormSubmit = useCallback(
+        (event_: React.BaseSyntheticEvent): void => {
+            void handleSubmit((data) => {
+                if (Object.keys(errors).length === 0) {
```

Comment:
this check is unnecessary since this callback will be called only if validation is successful